packages/core/src/util/txInspector.ts

```diff
@@ -167,16 +169,18 @@ export const getCertificatesByType = (
 ) => {
   if (!tx.body.certificates || tx.body.certificates.length === 0) return [];
   const certificates = certificateTypes
-    ? tx.body.certificates?.filter((certificate) => certificateTypes.includes(certificate.__typename))
+    ? tx.body.certificates?.filter((certificate) => isCertType(certificate, certificateTypes))
     : tx.body.certificates;
 
   return certificates.filter((certificate) => {
-    if ('stakeCredential' in certificate && certificate.stakeCredential) {
+    if (isCertType(certificate, StakeCredentialCertificateTypes)) {
       const credHash = Crypto.Ed25519KeyHashHex(certificate.stakeCredential.hash);
       return rewardAccounts.some((account) => RewardAccount.toHash(account) === credHash);
     }
 
-    if ('poolParameters' in certificate) return rewardAccounts.includes(certificate.poolParameters.rewardAccount);
+    if (isCertType(certificate, [CertificateType.PoolRegistration]) && 'poolParameters' in certificate)
```

Comment:
I missed this one. Added a fixup to address it
```diff
-    if (isCertType(certificate, [CertificateType.PoolRegistration]) && 'poolParameters' in certificate)
+    if (isCertType(certificate, [CertificateType.PoolRegistration]))

Reply:
Done here: [efd05a8](https://github.com/input-output-hk/cardano-js-sdk/pull/1074/commits/efd05a8d06766935d33d9a2c6e5ea726c9c5b29c)